lightning_action/api/video_model.py

```
@@ -0,0 +1,470 @@
+        """
+        videos_dir = Path(videos_dir)
+        labels_dir = Path(labels_dir)
+        output_dir = Path(output_dir)
+
+        if self.model is None:
+            raise ValueError('Model must be trained or loaded before prediction')
+
+        # Validate output_file usage
+        if output_file is not None and (expt_ids is not None and len(expt_ids) > 1):
+            raise RuntimeError(
+                'Can only supply `output_file` when specifying a single expt_id'
+            )
+
+        # Discover videos if not specified
+        if expt_ids is None:
+            expt_ids = [f[:-4] for f in os.listdir(videos_dir) if f.endswith('.mp4')]
+
+        num_videos = len(expt_ids)
+        if num_videos == 0:
+            print("No videos to predict")
+            return
+
+        # Validate that all required files exist
+        missing_files = []
+        for expt_id in expt_ids:
+            video_path = videos_dir / f'{expt_id}.mp4'
+            label_path = labels_dir / f'{expt_id}.npy'
+            if not video_path.exists():
+                missing_files.append(f"Video missing: {video_path}")
+            if not label_path.exists():
+                missing_files.append(f"Label missing: {label_path}")
+        
+        if missing_files:
+            raise FileNotFoundError(
+                f"Missing files for prediction:\n" + "\n".join(missing_files)
+            )
+
+        # Configure trainer for prediction
+        training_config = self.config.get('training', {})
+        batch_size = training_config.get('batch_size', 1)
+        device = training_config.get('device', 'cpu')
+        num_gpus = torch.cuda.device_count()
+        
+        # Use single GPU for prediction (simpler and more reliable)
+        if device == 'gpu' and num_gpus > 0:
+            accelerator = 'gpu'
+            devices = 1
+        else:
+            accelerator = 'cpu'
+            devices = 1
```

I think we want to throw an error here instead, I've tried several times in the past to ge DALI to work on CPU but it never has (despite the docs claiming otherwise). we should just force users to a GPU (especially with the ViT backbone, this is already slow even on a GPU)

---

https://github.com/paninski-lab/lightning-action/pull/24/commits/6b5df4fcf1cb6d7d75d96312859748daae01215e